generic/SmoothL1Criterion.c

```diff
@@ -0,0 +1,60 @@
+#ifndef TH_GENERIC_FILE
+#define TH_GENERIC_FILE "generic/SmoothL1Criterion.c"
+#else
+
+static int nn_(SmoothL1Criterion_updateOutput)(lua_State *L)
+{
+  THTensor *input = luaT_checkudata(L, 2, torch_Tensor);
+  THTensor *target = luaT_checkudata(L, 3, torch_Tensor);
+  int sizeAverage = luaT_getfieldcheckboolean(L, 1, "sizeAverage");
+  real sum;
+
+  sum = 0;
+  TH_TENSOR_APPLY2(real, input, real, target,
+                   real z = fabs(*input_data - *target_data);
```

Comment:
fabs will work correctly for double tensors. For float, you have to use fabsf.
These are the C99 prototypes for the function:
``C
     double fabs  (double x);
      float fabsf (float x);
``